### ISSUE TITLE
Android+Tailscale: No internet after moving from wifi to cellular

### ISSUE THREAD

**Issue by JustDoneNow** [tailscale/tailscale#11613]
### What is the issue?

Reference is https://github.com/tailscale/tailscale/issues/10107

Issue is closed but problem is not solved for android.

After moving out of range of my WIFI, the phone is connected to cellular. The phone and Tailscale say they are connected, but there is no internet. Can be resolved by either: Airplane mode on/off OR Tailscale app Active->Stop->Active

### Steps to reproduce

I'm at home connected to WIFI. Tailscale says it's connected. VPN and regular internet is working fine on the phone (Pixel 6, Pixel 7, Android 13 & 14)

When I leave home, WIFI goes out of range and the phone switches to cellular. The phone says 5G data is connected. The Tailscale app say it's active & connected.

If I try browsing to a website in Chrome, eventually you get the error "This site can't be reached whatever.com's server IP address could not be found. DNS_PROBE_FINISHED_BAD_CONFIG"

Internet can be restored by toggling Airplane mode or toggling Talescale from Active to Stop and back.

The problem can sometimes be reproduced by toggling WIFI state Android while connected to Tailscale

### Are there any recent changes that introduced the issue?

Have only been using Tailscale for a few months. Pretty sure it's always happened.

### OS

Android

### OS version

13 & 14

### Tailscale version

1.60.1

### Other software

Nothing else unusual. Two phones owned by different people - Pixel 6 & Pixel 7. The problem existed on Android 13 and still happens on Android 14.

Since the problem happens when away from home / not connected to WIFI and affects all network operations whether VPN or not, the rest of my home network is probably not relevant.

### Bug report

BUG-e2f204f87c09b3c608da1409f2f47ba74b63abe363dcb799f5fc588574f3a320-20240404171912Z-bfef24c4f8de03d8

BUG-e2f204f87c09b3c608da1409f2f47ba74b63abe363dcb799f5fc588574f3a320-20240404171952Z-17d05ad4d6ef340d

**Comment by carefulcomputer** [comment]
same issue here.

**Comment by JustDoneNow** [comment]
Added bug report

**Comment by starfishpatkhoo** [comment]
I had this problem also, basically apps on my phone would have a lot of difficulty connecting to the Internet when tailscale was connected. Sometimes it worked, sometimes it was just incredibly slow.

TL;DR - on Samsung phones (One UI 6), go to `Settings` > `Connections` > `More Connection Settings` > `Private DNS` > `Off` 

I think the default is `Automatic` and this would muck things up.

Once I changed it to `Off` everything works fine, on WiFi and on Cellular, and moving in and out of the two. Maybe this (while not strictly a Tailscale issue), should at least be noted in documentation for other users..

**Comment by carefulcomputer** [comment]
@starfishpatkhoo , in my case, when i turned the private dns from automatic to off, the tailscale stopped working. i couldn't access my network from my phone at all, even using IP addresses.

**Comment by JustDoneNow** [comment]
> I had this problem also, basically apps on my phone would have a lot of difficulty connecting to the Internet when tailscale was connected. Sometimes it worked, sometimes it was just incredibly slow.
> 
> TL;DR - on Samsung phones (One UI 6), go to `Settings` > `Connections` > `More Connection Settings` > `Private DNS` > `Off`
> 
> I think the default is `Automatic` and this would muck things up.
> 
> Once I changed it to `Off` everything works fine, on WiFi and on Cellular, and moving in and out of the two. Maybe this (while not strictly a Tailscale issue), should at least be noted in documentation for other users..

Seems to work for me also, thanks for the hint.
Let's See how works in the next days.

**Comment by starfishpatkhoo** [comment]
> @starfishpatkhoo , in my case, when i turned the private dns from automatic to off, the tailscale stopped working. i couldn't access my network from my phone at all, even using IP addresses.

Hmm, if IP address isn't working, that's maybe something else? Maybe you need to reauth, or reboot, or such.. Check if the VPN is active etc..

**Comment by carefulcomputer** [comment]
i deleted docker container's mounted volume and restarted and it started to work. thanks.

**Comment by JustDoneNow** [comment]
> i deleted docker container's mounted volume and restarted and it started to work. thanks.

**Comment by JustDoneNow** [comment]
> i deleted docker container's mounted volume and restarted and it started to work. thanks. I hope DNS issue does get resolved though. I am going to try the auto DNS turned off.

The issue has nothing to do with docker.
I also got the it with auto DNS deactivated.
So that's not a solution to the problem either.

**Comment by NimecShady** [comment]
Just wanted to chime in as well and say both my phone and my wifes phone (Pixel7 & Pixel8) both also lose internet connectivity when leaving our house and going from Wifi to cellular.  The only way we have found to fix it is to turn tailscale off and back on again through the app.  My friend using an iPhone does not have similar experience so sounds like an android specific issue?

**Comment by kari-ts** [comment]
Hi! We recently [fixed](https://github.com/tailscale/tailscale-android/commit/a6bc2244b66fddf65f6e36b4fc5202f7589cd5c0) a DNS issue. We'd really appreciate your help testing it out to see if it fixes your issue! Please become an open tester to try out our beta [here](https://play.google.com/store/apps/details?id=com.tailscale.ipn). If you prefer to wait, this will be out as part of the 1.66 release.

**Comment by JustDoneNow** [comment]
> Hi! We recently [fixed](https://github.com/tailscale/tailscale-android/commit/a6bc2244b66fddf65f6e36b4fc5202f7589cd5c0) a DNS issue. We'd really appreciate your help testing it out to see if it fixes your issue! Please become an open tester to try out our beta [here](https://play.google.com/store/apps/details?id=com.tailscale.ipn). If you prefer to wait, this will be out as part of the 1.66 release.

Is this fix included in 1.65.95-t71e9258ad-g5ea9a50ba57? I'm already a beta tester and will check out the behavior.

**Comment by codethief** [comment]
Possibly related tickets/conversations: 
- https://github.com/tailscale/tailscale/issues/6516#issuecomment-1546866703
- https://github.com/tailscale/tailscale/issues/9492

**Comment by codethief** [comment]
@kari-ts 
> If you prefer to wait, this will be out as part of the 1.66 release.

Which version exactly is it part of? I don't see anything in the changelogs for the 1.66.x releases so far.

**Comment by kari-ts** [comment]
[This](https://github.com/tailscale/tailscale-android/pull/340) is in 1.66.3

**Comment by codethief** [comment]
@kari-ts Thanks, I've been running v1.66.3 for the past two days and so far I haven't had any connection issues anymore.

**Comment by kari-ts** [comment]
@codethief that's fantastic, so happy to hear! Thank you for the followup. Closing this as done, please get in touch if you find any other issues.

**Comment by JustDoneNow** [comment]
Works very well, thanks for your work 👍
